### ISSUE TITLE
upgrade support (druid)

### ISSUE THREAD

**Issue by smildlzj** [metabase/metabase#7300]
i had v0.27.2 upgrade to v0.28.2

i saw console output like this:
```
INFO sync-metadata.fields :: Database type of Table 13 'web-im@performance' Field 'ttfb' has changed from '?' to 'LONG'.
INFO sync-metadata.fields :: Base type of Table 13 'web-im@performance' Field 'ttfb' has changed from ':type/Text' to ':type/Integer'.
```

my segment like this:
```
"dimensions" : [
    {
      "type": "long",
      "name": "ttfb"
    }
  ]
}
```
and when i get result from question, something go wrong
```
WARN druid.query-processor :: WARNING: Filtering only works on dimensions! 'ttfb' is a metric. Ignoring :> filter
```

![5c78ee9297b0dab0f8a96f59ea23e146c06274af_1_689x414](https://user-images.githubusercontent.com/1638850/38484128-1905559a-3c08-11e8-822d-cbdabd1e86a4.png)

**Comment by smildlzj** [comment]
can anyone help ? pls....

**Comment by dgregoire** [comment]
I've also run into this issue. Additional info:

If a data type in a druid schema is actually long or float

```
	      "dimensionsSpec" : {
	        "dimensions": [
	          { "name" : "userid", "type" : "long" },
```

it gets automatically treated as a metric, generating filtering errors
```
05-07 02:55:45 WARN druid.query-processor :: WARNING: Filtering only works on dimensions! 'userid' is a metric. Ignoring filter.
```

seems like this code makes that assumption

https://github.com/metabase/metabase/blob/d7c6d1f41e8bd6db3b418e6e92aba87466c2a868/modules/drivers/druid/src/metabase/driver/druid/query_processor.clj#L129

is there any work around to this?

**Comment by aoeiuvb** [comment]
I also encountered this problem, filtering does not work, how do you solve

**Comment by cparaskeva** [comment]
Any updates regarding the issue?

**Comment by flamber** [comment]
This was fixed in 0.35.0 with a complete rewrite of Druid query from Select to Scan.
